cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/type/DatasetTypeManager.java

```diff
@@ -376,6 +378,27 @@ private void deployDefaultModules() {
     }
   }
 
+  private void deleteSystemModules() {
+    try {
+      mdsDatasets.execute(new TxCallable<MDSDatasets, Void>() {
+        @Override
+        public Void call(MDSDatasets context) throws Exception {
+          DatasetTypeMDS typeMDS = context.getTypeMDS();
+          Collection<DatasetModuleMeta> allDatasets = typeMDS.getModules(Constants.SYSTEM_NAMESPACE_ID);
+          for (DatasetModuleMeta ds : allDatasets) {
+            if (ds.getJarLocation() == null) {
```

Comment:
I see. Cool. Also, any upgrade implications? This change will probably not be able to delete the system modules from earlier because the key will be old-style - *m_<module_name>*? Does it make sense then if the upgrade tool ignores system modules? Only migrates over non-system ones - the ones with non-null ``jarLocation``s? They will all go to a different table from before anyway.